app/extend/context.js

```diff
@@ -50,7 +51,7 @@ module.exports = {
 
     this[HAS_CONSUMED] = true;
     const parseOptions = {};
-    Object.assign(parseOptions, this.app.config.multipartParseOptions, options);
+    deepmerge.all([parseOptions, this.app.config.multipartParseOptions, options]);
```

Comment:
use `extend2` instead of `deepmerge`, which is used by [egg](https://github.com/eggjs/egg/blob/master/package.json#L43)

check if options not presents, no need to do this deep copy action.